develop: Improve integ tests code to be compatible with hpc6id

enrico-usai opened the pull request:
### 1. Reduce threshold for hpc6id EFA test to be compatible with Ubuntu 20
I did multiple tests and Ubuntu 20 I obtained a value of 23057.79

### 2. Increase number of MaxvCpus to fit with instances with more vcpus

The test will be ready if in future instances like hpc6id will be supported by AWS Batch.

### 3. Make pcluster configure test prompts compatible with instances with EFA

If the instance has EFA support (like hpc6id) the prompt will ask to enable EFA and if answering in a positive way, will ask for placement group too.

In this way the code is generic enough to support both the cases.
I tested it using hpc6id and c5.xlarge as instances.

### Tests
* Tested running the tests with hpc6id as instance.
* configure change tested with c5.large as instance too.

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Commit:
Reduce threshold for hpc6id EFA test to be compatible with Ubuntu 20

I did multiple tests and Ubuntu 20 I obtained a value of 23057.79

Signed-off-by: Enrico Usai <usai@amazon.com>

Commit:
Increase number of MaxvCpus to fit with instances with more vcpus

The test will be ready if in future instances like hpc6id will be supported by AWS Batch.

Signed-off-by: Enrico Usai <usai@amazon.com>

eantonin commented on tests/integration-tests/tests/efa/test_efa.py:
```diff
@@ -213,7 +213,7 @@ def _test_osu_benchmarks_multiple_bandwidth(
         "p4de.24xlarge": 30000,  # Equivalent to a theoretical maximum of a single 240Gbps card
         # 2 up to 170 Gbps NICS -> declared NetworkPerformance 200 Gbps
-        "hpc6id.32xlarge": 23500,  # Equivalent to a theoretical maximum of a single 188Gbps card
+        "hpc6id.32xlarge": 23000,  # Equivalent to a theoretical maximum of a single 188Gbps card
```
nitpick, the comment is now incorrect, should be `184Gbps`

eantonin commented on tests/integration-tests/tests/multiple_nics/test_multiple_nics/test_multiple_nics/pcluster.config.yaml:
```diff
@@ -1,7 +1,7 @@
   Os: {{ os }}
 HeadNode:
-  InstanceType: {{ instance }}
+  InstanceType: c5.xlarge
```
question, blocking: (here and below) what do this imply? Seems to me that this will reduce the instance types tested in the headnode. In my opinion we should:
* Do this change locally to test hpc6id (as I assume you did)
* Not merge it
* Once we support the hpc6id in the headnode we can add it to the list of instances tested

enrico-usai replied:
wrong commit, I'm going to remove it.

Commit:
Make pcluster configure test prompts compatible with instances with EFA

If the instance has EFA support (like hpc6id) the prompt will ask to enable EFA
and if answering in a positive way, will ask for placement group too.

In this way the code is generic enough to support both the cases.
I tested it using hpc6id and c5.xlarge as instances.

Signed-off-by: Enrico Usai <usai@amazon.com>

eantonin reviewed (approved):
LGTM
